src/v/kafka/client/consumer.cc

```diff
@@ -119,14 +125,24 @@ void consumer::start() {
 }
 
 ss::future<> consumer::stop() {
-    { auto t = std::move(_heartbeat_timer); }
+    vlog(kclog.info, "Consumer: {}: stop", *this);
+    {
+        auto h = std::move(_heartbeat_timer);
+        auto i = std::move(_inactive_timer);
+    }
```

Comment:
what's the purpose of the moves here?

Reply:
got it. ideally shortly after the stop method is called the call instance is destroyed (along with the timers) so the lifetimes are all aligned. but assuming that isn't an option then I'd just prefer `std::exchange(timer, {})` here with a comment that it both cancels the timer when local scope exits and most importantly, the lifetime of the shared pointer is managed.

Reply:
Ahh, but it won't be destroyed because there's a `shared_ptr` in the callback. `std::exchange(timer, {})` and a comment is perfect, thanks.

Reply:
Actually, `std::exchange` requires the type to be assignable, which `ss::timer` is not.

Reply:
ahh. then I think I'd vote for

```
_heartbeat_timer.cancel();
_heartbeat_timer.set_callback([](){});
_inactive_timer.cancel();
_inactive_timer.set_callback([](){});
```

plus a comment.

Reply:
it is very pedantic to avoid std::move in this case, but unlike std::exchange for instance, std::move doesn't guarantee anything is moved (though in practice for std::timer it certainly is).

Reply:
No it's fine, I understand. It probably took me a while to figure out what was needed in the first place. Being explicit and helping with a comment makes total sense.